Uploading Design of 2 DC-130+ GHz TIA for April 07 OpenMPW

Ikram-rs22 opened the pull request:
All the required documentation are provided accordingly. If needed, further updates will be added

Ikram-rs22 committed:
initial testing

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
2nd test

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
3rd test

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
Updating files

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
validation update

Ikram-rs22 committed:
Update readme.md

Ikram-rs22 committed:
Update readme.md

Ikram-rs22 committed:
Update readme.md

Ikram-rs22 committed:
testing rst

Ikram-rs22 committed:
Merge branch 'main' of https://github.com/Ikram-rs22/TO_Apr2025

Ikram-rs22 committed:
rst update

Ikram-rs22 committed:
rst image

Ikram-rs22 committed:
rst image test

Ikram-rs22 committed:
rst image test 2

Ikram-rs22 committed:
Update validation.rst

Ikram-rs22 committed:
Add files via upload

Ikram-rs22 committed:
rst test

Ikram-rs22 committed:
gds name change

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
rst image width

Ikram-rs22 committed:
Updated gds, schematic and s2p files

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
lvs image

Ikram-rs22 committed:
800 width

Ikram-rs22 committed:
height

Ikram-rs22 committed:
width change

Ikram-rs22 committed:
aspect ratio change

Ikram-rs22 committed:
2nd aspect

Ikram-rs22 committed:
final aspect

Ikram-rs22 committed:
atif bhai check

Ikram-rs22 committed:
drc 2 check

Ikram-rs22 committed:
my gds

Ikram-rs22 committed:
Adding files for documentation

Ikram-rs22 committed:
change

Ikram-rs22 committed:
Merge branch 'main' of https://github.com/Ikram-rs22/TO_Apr2025

Ikram-rs22 committed:
Image modify

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
changing image

Ikram-rs22 committed:
image check

Ikram-rs22 committed:
changing some aspects

Ikram-rs22 committed:
test text

Ikram-rs22 committed:
Validation updated

Ikram-rs22 committed:
image edit

Ikram-rs22 committed:
changed gds

Ikram-rs22 committed:
esbonio fail

Ikram-rs22 committed:
Updated gds

Ikram-rs22 committed:
flow chart

Ikram-rs22 committed:
Add files via upload

Ikram-rs22 committed:
1200 width

Ikram-rs22 committed:
Merge branch 'main' of https://github.com/Ikram-rs22/TO_Apr2025

Ikram-rs22 committed:
resize

Ikram-rs22 committed:
800 width

Ikram-rs22 committed:
Add files via upload

Ikram-rs22 committed:
flowchart

Ikram-rs22 committed:
Merge branch 'main' of https://github.com/Ikram-rs22/TO_Apr2025

Ikram-rs22 committed:
flowchart rearranged

Ikram-rs22 committed:
deletion

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
Total update

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
image resize

Ikram-rs22 committed:
2nd resize

Ikram-rs22 committed:
image resize

Ikram-rs22 committed:
image upload

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
schematic upload image

Ikram-rs22 committed:
schematic resize

Ikram-rs22 committed:
resizing still

Ikram-rs22 committed:
justify

Ikram-rs22 committed:
not justify

Ikram-rs22 committed:
image resized

Ikram-rs22 committed:
schematic deleted

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
Updated pins

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
Updated pins

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
Updated some image

Ikram-rs22 committed:
change

Ikram-rs22 committed:
Update again pin

Ikram-rs22 committed:
em sim img

Ikram-rs22 committed:
checking

Ikram-rs22 committed:
test

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
update

Ikram-rs22 committed:
Almost done

Ikram-rs22 committed:
image resize

Ikram-rs22 committed:
qucs-s

Ikram-rs22 committed:
specification

Ikram-rs22 committed:
Update index.rst

Ikram-rs22 committed:
Final

Ikram-rs22 committed:
Last almost

Ikram-rs22 committed:
Done

Ikram-rs22 committed:
Update designdata.rst

Ikram-rs22 committed:
PA DRC Check

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
PA test

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
Testing DRC

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
Merge remote-tracking branch 'origin/main'

Ikram-rs22 committed:
PA Check

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
PA_Check_done

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
drc/results added

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
removed Result

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
GDS and Density updated

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
Update conf.py

Author & Design name

Ikram-rs22 committed:
Updated the gds and documentation

Ikram-rs22 committed:
Merge branch 'main' of https://github.com/Ikram-rs22/TO_Apr2025

Ikram-rs22 commented:
The gds has been updated. Both the minimal and maximal drc are clean. All required documentation, circuit testbench have been provided. Will be updated further if needed to be from IHP's end.

Ikram-rs22 committed:
A 2nd version test

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
Design Completed

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
Fixing minor errors

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
Design Sign-off

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
Update validation.rst

Image renamed

Ikram-rs22 committed:
Padframe modified sign-off

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
A 2nd Design

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
All doc of 2nd Design Uploaded

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 commented:
A 2nd design with required documentation and files have been uploaded. DRC & LVS is clean for both designs.

Ikram-rs22 committed:
Config file and S2P updated

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
Overall Design Check

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 commented:
GSG pads for RF measurements only; not used for bonding or standard probing—no dfpad layer applied intentionally

KrzysztofHerman commented:
Dear Contributor, 
since we are closing the submission please change the final GDS name adding the following prefix to the file 
`FMD_QNC_03_`

Ikram-rs22 commented:
Do I need to change the name of both the gds to FMD_QNC_03?? @KrzysztofHerman

Ikram-rs22 committed:
Updating GDS Names

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
Changing names

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
Deleting Duplicate

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 commented:
@KrzysztofHerman what should be the GDS names? BOTH FMD_QNC_03?

Ikram-rs22 committed:
Design_2 folder completed

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
Uploading Different Designs, names changed

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 commented:
@KrzysztofHerman as there is no clarification from your end, I've renamed both my designs with the prefix FMD_QNC_03. Please kindly don't hold me accountable for any future inconveniences. 

Regards, Ikram

KrzysztofHerman commented:
please provide names as follows: `FMD_QNC_03a`, `FMD_QNC_03b`

Ikram-rs22 commented:
Thank you

Ikram-rs22 committed:
Renamed GDS with FMD_Prefix

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 committed:
Deleted Duplicate files

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 commented:
All files have been renamed with the FMD_QNC_03 prefix.

Ikram-rs22 committed:
Labeling pads of gds

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

KrzysztofHerman commented:
@Ikram-rs22 
I have verified your design using internal DRC and it is clean and almost ready to be accepted. I have to ask you for the following modifications:
1) Change the top level cell name for something meaningful, it could be also the suggested prefix `FMD_QNC_03a` and `FMD_QNC_03b` respectively.  Keep the name short < 20 characters.
2) Due to internal issues we have to ask you to open the file in klayout and save it again using `File->Save As..` and provide the following options: 

![image](https://github.com/user-attachments/assets/9c03da59-d0fa-4c82-b2a2-366f080b93f8)

3) We have to ask you for some contact data and the shipping address. 

4) Please download the Hiring Agreement from [here](https://ihp-open-ip.readthedocs.io/en/latest/_downloads/c82c58426246cc84f3bbce282975b255/samples.rtf) and sign it. If you are with an university please ask for signature some representative: your advisor, group leader, departments head.  

5) The data mentioned in the points 3-4 you can send to the following [email](openpdk@ihp-microelectronics.com). Please mention this tape out name `TO_Apr2025` in the email title.

Ikram-rs22 committed:
Top level Cell Names & GDS options modified

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 commented:
@KrzysztofHerman I've renamed the top level cell names and also modified the GDS write options. Could you kindly check if the write options have been appropriated?? 

As for the contact and shipping address, I'm currently working on the hiring agreement. I'll let you know when the documents have been submitted.

KrzysztofHerman commented:
@Ikram-rs22 

After some closer look I have detected DRC violation in the design `FMD_QNC_03a_DC_to_130_GHz_TIA_Design.gds`. 
It is about `M1.k`. Please remove some fillers on metal1 and submit again.

Ikram-rs22 commented:
Could this be false positive?

Ikram-rs22 commented:
![Screenshot_20250409-174629.Chrome.png](https://github.com/user-attachments/assets/e8a22de6-3867-4312-81b6-b5626c1b068e)

The report shows there are no errors.

KrzysztofHerman commented:
@ikram 

After checking the designs, especially the method you have generated fillers I have to ask you to do the following:

Please read this [documentation](https://ihp-open-pdk-docu.readthedocs.io/en/latest/finishing/filler.html) and follow these steps:

1. remove all filler structures and all layers,
2. generate fillers using filler scripts  
![image](https://github.com/user-attachments/assets/e9a0a320-8878-41b0-8810-3d58c14fec14)
3. Run Density report and check the values. 
4. Run minimal DRC
5. Fix the metal densities in the case of density issues
6. Get familiar with github issues related to the Fill/Density checks.
7. Re-submit

We still have time to fix the issues and have successful submission.

Ikram-rs22 commented:
@KrzysztofHerman okay, I'll do it. I didn't use fill cells for metal density. Is this a problem? Only Active & Gat Poly fill cells have been used. Do I need to use the filler scripts for all the metal and activ/gat poly fills or only for active/gat poly fills.  And please kindly waive the drc error regarding the fill cell related are (e g. 800 x 800 um^2)

Ikram-rs22 committed:
Edited Filler

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 commented:
> @Ikram-rs22
> 
> After some closer look I have detected DRC violation in the design `FMD_QNC_03a_DC_to_130_GHz_TIA_Design.gds`. It is about `M1.k`. Please remove some fillers on metal1 and submit again.

I've deleted the metal 1 filler and re-ran the active/gatpoly filler scripts for FMD_QNC_03a_DC_to_130_GHz_TIA_Design.gds . Now only the 800 um X 800 um errors show. Also the density report is given below:
![density_report](https://github.com/user-attachments/assets/cc14c3f6-ac68-4c51-9d9f-50182d0031a6)
Please kindly inform me if this solves your issue @KrzysztofHerman . If this doesn't solve then I'll change both my gds accordingly. Thanks

KrzysztofHerman commented:
@Ikram-rs22 
please fill the design using the menu items. Try to trim it to be close to the center. Still we do not have 100% working flow to run this step and it is mandatory to submit the design.

Ikram-rs22 committed:
Re-filled densities and updated documentation

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 commented:
@KrzysztofHerman I've revised the gds as per your instructions. Deleted all the layers and fillers and ran the filler script for density requirements. The FMD_QNC_03b design shows 2 800 X 800 um window error, other than that all the drcs are clean. I've filled all the fillers from menu scripts. I've also updated the documentation and density report shows that density requirements are met. Please kindly confirm if any other modifications are to be made. 

**Note that Metal 1 layer is kept untouched because this serves as a global ground for RF design and is mandatory. RF shielding is provided by global metal 1 ground**

KrzysztofHerman commented:
@Ikram-rs22 

Your `FMD_QNC_03a_DC_to_130_GHz_TIA_Design.gds` still violates the M1 Max density rules. 
The density report shows `Metal1 %    = 58.182% Min = 35 Max = 60` Please try to keep it under 50%. 
Please fix and resubmit.
The second design `FMD_QNC_03b_` is already accepted for production.

Ikram-rs22 commented:
Okay, On it.

Ikram-rs22 committed:
Metal 1 Density of design 45-46%

Signed-off-by: Muhammad Ikram <ikramsultan00@gmail.com>

Ikram-rs22 commented:
@KrzysztofHerman I've updated the density of FMD_QNC_03a_DC_to_130_GHz_TIA_Design.gds to 45% and FMD_QNC_03b to 46%. Both designs have 2 drcs regarding the 800 um X 800 um fill area. please kindly check if this meets the requirement.

KrzysztofHerman commented:
@Ikram-rs22 

The designs are minimum DRC clean and ready for production.
Please take care about the Hiring Agreement and the contact data.

Ikram-rs22 commented:
Okay. Thanks a lot @KrzysztofHerman for your continuous guideline and effort. You've helped me a lot. Thanks to IHP GmBH as well for providing this opportunity. I'll inform you about the hiring agreement very soon.

Ikram-rs22 commented:
@KrzysztofHerman I've submitted the hiring agreement. Please kindly inform me if there is any problem regarding the hiring agreement or the designs.
Once again, thank you for your effort.

KrzysztofHerman commented:
@Ikram-rs22 


Congratulations!
The design was accepted for production:
```
File name: FMD_QNC_03a_DC_to_130_GHz_TIA_Design.gds
Top cell name: FMD_QNC_03a_TIA_1
Testfield: T586
Check Summe: ce47f6e4a8d93b250c7376467a8a129b
Sent: 11-04-2025, 10:47:32
registered size: (0.76 mm X 0.9 mm ) 
calculated size: (0.71 mm X 0.85 mm))
ID (use this as reference for questions): 54544


```
```
File name: FMD_QNC_03b_2nd_Design_3_Stage.gds
Top cell name: FMD_QNC_03b_TIA_2
Testfield: T586
Check Summe: 45282717612e7e51ac1200228b8c6679
Sent: 11-04-2025, 11:05:46
registered size: (0.9 mm X 0.78 mm ) 
calculated size: (0.85 mm X 0.73 mm))
ID (use this as reference for questions): 54548

```

Please do not change the GDS files anymore.
I will merge this PR to the main branch

Ikram-rs22 commented:
Thank you @KrzysztofHerman

KrzysztofHerman commented:
Hi, @Ikram-rs22 

I had to update your GDS. The issue was that you have used some custom method to generate 
Active and GatPoly fillers, what resulted in non-acceptable design. I had to regenerate it and resubmit.   
We will develop a DRC check for this case. 
Please sync your GDS with the updated one.

Ikram-rs22 commented:
Okay, was it for both the designs?

On Wed, 30 Apr 2025, 6:18 pm Krzysztof Herman, ***@***.***>
wrote:

> *KrzysztofHerman* left a comment (IHP-GmbH/TO_Apr2025#3)
> <https://github.com/IHP-GmbH/TO_Apr2025/pull/3#issuecomment-2841791638>
>
> Hi, @Ikram-rs22 <https://github.com/Ikram-rs22>
>
> I had to update your GDS. The issue was that you have used some custom
> method to generate
> Active and GatPoly fillers, what resulted in non-acceptable design. I had
> to regenerate it and resubmit.
> We will develop a DRC check for this case.
> Please sync your GDS with the updated one.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/IHP-GmbH/TO_Apr2025/pull/3#issuecomment-2841791638>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQQOWEZRP66H7HA3QAY2DET24C5RXAVCNFSM6AAAAABZHSAZC2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQNBRG44TCNRTHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Ikram-rs22 commented:
Could you kindly tell me how do I sync my gds with the updated design? As I'm not too adept in github.

Ikram-rs22 commented:
@KrzysztofHerman have the designes been approved for fabrication??

Ikram-rs22 commented:
@KrzysztofHerman are the designs drc clean for fabrication?

KrzysztofHerman commented:
Yes, we have fixed it and it was accepted. 

```
File name: FMD_QNC_03a_DC_to_130_GHz_TIA_Design_1.gds
Top cell name: FMD_QNC_03a_TIA_1
Testfield: T586
Check Summe: 28293cf48b1db34439d8cb5589fb89fb
Sent: 30-04-2025, 10:08:03
registered size: (0.76 mm X 0.9 mm ) 
calculated size: (0.71 mm X 0.85 mm))
ID (use this as reference for questions): 54626


```


```
File name: FMD_QNC_03b_2nd_Design_3_Stage_1.gds
Top cell name: FMD_QNC_03b_TIA_2
Testfield: T586
Check Summe: e38b9044f2ae721b0859cd0301f0dc7c
Sent: 30-04-2025, 10:08:03
registered size: (0.9 mm X 0.78 mm ) 
calculated size: (0.85 mm X 0.73 mm))
ID (use this as reference for questions): 54627
```
